sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/IncrementalExecution.scala

```diff
@@ -347,6 +347,27 @@ class IncrementalExecution(
           eventTimeWatermarkForEviction = inputWatermarkForEviction(m.stateInfo.get)
         )
 
+      case UpdateEventTimeColumnExec(eventTime, delay, _,
+        SerializeFromObjectExec(serializer,
+        TransformWithStateExec(keyDeserializer, valueDeserializer, groupingAttributes,
+        dataAttributes, statefulProcessor, timeMode, outputMode,
+        keyEncoder, outputAttr, stateInfo, batchTimestampMs, _,
+        _, child, isStreaming, hasInitialState,
+        initialStateGroupingAttrs, initialStateDataAttrs,
+        initialStateDeserializer, initialState))) if stateInfo.isDefined =>
+
+        val eventTimeWatermarkForLateEvents = inputWatermarkForLateEvents(stateInfo.get)
```

Comment:
Could we add a comment for this block in general ?

Reply:
yeah. Done